controllers/controlplaneupgrade_controller.go

```diff
@@ -310,6 +316,30 @@ func (r *ControlPlaneUpgradeReconciler) updateKubeadmConfig(ctx context.Context,
 	return nil
 }
 
+func (r *ControlPlaneUpgradeReconciler) updateInfraMachine(ctx context.Context, log logr.Logger, kcpSpec *controlplanev1.KubeadmControlPlaneSpec, machine *clusterv1.Machine) error {
```

Comment:
do we have to give rbac permissions to update/patch infra machines

Reply:
Going to test that, but I feel like our controller probably has permissions somewhere for all the infra machines?

Reply:
Based on this, have to explicitly specify tinkerbellmachines and vspheremachines for now: https://kubernetes.io/docs/reference/access-authn-authz/rbac/#referring-to-resources